### PR TITLE
Add new Grand Spectrum mods and change Grand Spectrum parsing to generic one

### PR DESCRIPTION
**Pull request by deathbeam**
Adds support for frenzy, power, life and ailment Grand Spectrum

Signed-off-by: Tomas Slusny <slusnucky@gmail.com>

## Example POB

https://pobb.in/inf9BcVDcjdS

### CODE REVIEW

File: src/Data/Uniques/jewel.lua [hunk 1]
```diff
@@ -247,10 +247,12 @@ Variant: Pre 3.0.0
 Variant: Pre 3.10.0
 Variant: Current - Crit Chance
 Variant: Current - Min Endurance Charge
+Variant: Current - Min Power Charge
 {variant:1}Gain 15 Mana per Grand Spectrum
 {variant:2}Gain 30 Mana per Grand Spectrum
 {variant:3}25% increased Critical Strike Chance per Grand Spectrum
 {variant:4}+1 to Minimum Endurance Charges per Grand Spectrum
```
Comment (learn2draw) [inline src/Data/Uniques/jewel.lua:254]: The min endurance charge on is the crimson base, could you move htis one? 
image

File: src/Data/Uniques/jewel.lua [hunk 2]
```diff
@@ -259,10 +261,12 @@ Variant: Pre 3.0.0
 Variant: Pre 3.10.0
 Variant: Current - Elemental Resistances
 Variant: Current - Minion Crit Multi
+Variant: Current - Maximum Life
 {variant:1}Gain 75 Armour per Grand Spectrum
 {variant:2}Gain 200 Armour per Grand Spectrum
 {variant:3}+7% to all Elemental Resistances per Grand Spectrum
 {variant:4}Minions have +10% to Critical Strike Multiplier per Grand Spectrum
```
Comment (learn2draw) [inline src/Data/Uniques/jewel.lua:268]: Could you move the minion crit to a cobalt

**Comment by deathbeam** [comment]
Moved both

File: src/Classes/Item.lua [hunk 3]
```diff
@@ -1161,6 +1161,10 @@ function ItemClass:BuildModListForSlotNum(baseList, slotNum)
 			flaskData[value.key] = value.value
 		end
 	elseif self.type == "Jewel" then
+		if string.find(self.name, "Grand Spectrum") then
```
Comment (PJacek) [inline src/Classes/Item.lua:1164]:
```suggestion
		if self.name:match("Grand Spectrum") then
```
Avoid accessing the global `string`

Comment (deathbeam) [reply src/Classes/Item.lua]: Updated

**Review by LocalIdentity — APPROVED** [review]

**Pull request merged**